mongo/errors.go

```diff
@@ -35,6 +35,9 @@ var ErrNilValue = errors.New("value is nil")
 // ErrEmptySlice is returned when an empty slice is passed to a CRUD method that requires a non-empty slice.
 var ErrEmptySlice = errors.New("must provide at least one element in input slice")
 
+// ErrInvalidLoggerOptions is returned when an invalid logger is provided.
+var ErrInvalidLoggerOptions = errors.New("invalid logger options")
+
```

Comment:
Users are unlikely to handle `Client` configuration errors with runtime logic, so we don't typically need to support programmatic error checking for those errors (users just need to get a human-readable error message). This should be removed.
```suggestion
```